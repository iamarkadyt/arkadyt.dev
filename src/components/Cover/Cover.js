import React from 'react';
import classes from './Cover.module.css';
import { CoverCtx } from '../../hocs/withCoverCtx';

const Cover = () => {
    return (
        <CoverCtx.Consumer>
            {context => {
                const cls = [classes.container, context.coverLifted ? classes.lifted : ''].join(' ');
                return <div className={cls}>
                    <img src='http://placehold.it/1920x1080' alt="" />
                </div>;
            }}
        </CoverCtx.Consumer>
    );
}
export default Cover;
