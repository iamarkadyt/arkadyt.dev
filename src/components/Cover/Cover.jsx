import React from 'react';
import { connect } from 'react-redux';
import * as actionTypes from '../../store/actions';

import classes from './Cover.module.css';
import { CoverCtx } from '../../hocs/withCoverCtx';

export class Cover extends React.Component {
    render() {
        return (
            <CoverCtx.Consumer>
                {context => {
                    const cls = [classes.container, context.coverLifted ? classes.lifted : ''].join(' ');
                    return <div className={cls}>
                        <img src='http://placehold.it/1920x1080' alt="" />
                    </div>;
                }}
            </CoverCtx.Consumer>
        );
    }

    componentDidMount() {
        if (!this.props.componentLoaded) {
            this.props.onComponentLoaded();
        }
    }
}

const mapStateToProps = state => {
    return {
        componentLoaded: state.coverLoaded
    }
}

const mapDispatchToProps = dispatch => {
    return {
        onComponentLoaded: () => dispatch({ type: actionTypes.COVER_LOADED })
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Cover);
