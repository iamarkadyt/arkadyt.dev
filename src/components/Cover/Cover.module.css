.container {
    position: fixed;
    top: 0;
    left: 0;
    z-index: 4;
    width: 100vw;
    height: 100vh;
    transition: top 1s;
    background-image: url(./Cover-BG.jpg);
    /* background: linear-gradient(to top, rgba(47,54,64,0.95), rgba(72,126,176,0.68)); */
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
}

.container.lifted {
    top: -100vh;
}

@font-face {
    font-family: 'Cover Header';
    src: url(../../fonts/engebrechtrebd.ttf) format('truetype');
}

@font-face {
    font-family: 'Paragraph';
    src: url(../../fonts/dustismo.ttf) format('truetype');
}

.title {
    text-align: center;
    color: white;
    text-transform: uppercase;
    background-color: rgba(0, 0, 0, 0.5);
    padding: 10px 40px;
}

.title h1 {
    font-family: 'Cover Header', sans-serif;
    font-weight: 300;
    font-size: 4em;
    margin: 0;
}

.title p {
    font-family: 'Paragraph', sans-serif;
    font-weight: 300;
    margin: 0;
}