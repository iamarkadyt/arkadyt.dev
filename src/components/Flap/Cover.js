import React from 'react';
import classes from './Cover.module.css';
import { LiftedCtx } from '../../App';

const Cover = () => {
    return <LiftedCtx.Consumer>
        {context => {
            const cls = [classes.container, context.coverLifted ? classes.lifted : ''].join(' ');
            return <div className={cls}>
                <img src='http://placehold.it/1920x1080' alt="" />
            </div>;
        }}
    </LiftedCtx.Consumer>;
}
export default Cover;
