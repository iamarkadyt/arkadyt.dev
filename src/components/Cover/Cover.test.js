import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import classes from './Cover.module.css';
import Cover from './Cover';

Enzyme.configure({ adapter: new Adapter() })

describe('<Cover/>', () => {
    it('matches snapshot.', () => {
        const outer = shallow(<Cover />);
        const Children = outer.props().children({coverLifted: true});
        const wrapper = shallow(Children);
        expect(wrapper).toMatchSnapshot();
    });

    it('has .lifted class when coverLifted (coming from CoverCtx context) is true.', () => {
        const outer = shallow(<Cover />);
        const Children = outer.props().children({coverLifted: true});
        const wrapper = shallow(Children);
        expect(wrapper.find('.lifted')).toHaveLength(1);
    });

    it('does not have .lifted class when coverLifted is false.', () => {
        const outer = shallow(<Cover />);
        const Children = outer.props().children({coverLifted: false});
        const wrapper = shallow(Children);
        expect(wrapper.find('.lifted')).toHaveLength(0);
    });
});