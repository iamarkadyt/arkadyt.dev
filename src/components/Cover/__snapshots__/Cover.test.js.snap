// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cover/> matches snapshot. 1`] = `
<div
  className="container lifted"
>
  <div
    className="title"
  >
    <h1>
      Arkady Titenko
    </h1>
    <p>
      Software Engineer, Web Developer, Game Designer
    </p>
  </div>
</div>
`;
